src/api/java/io/github/cvc5/api/Datatype.java

```diff
@@ -24,20 +24,16 @@
   Datatype(Solver solver, long pointer)
   {
     super(solver, pointer);
+    solver.addAbstractPointer(this);
```

Comment:
Can we move call into the `AbstractPointer` base class?

Reply:
Done.